src/materialized/src/http/prof.rs

```diff
@@ -130,6 +142,30 @@ mod enabled {
                     .body(Body::from(s))
                     .unwrap())
             }
+            "flamegraph" => {
+                let mut borrow = prof_ctl.lock().expect("Profiler lock poisoned");
+                let f = borrow.dump()?;
+                let r = BufReader::new(f);
+                let stacks = parse_jeheap(r)?;
+                let collated = collate_stacks(stacks);
+                let data_json = RefCell::new(String::new());
+                collated.dfs(
+                    |node| {
+                        data_json.borrow_mut().push_str(&format!(
+                            "{{\"name\": \"{}\",\"value\":{},\"children\":[",
+                            node.name, node.weight
+                        ));
```

Comment:
If there are lots of collated elements you could `use std::fmt::Write` and just `write!` to the data json, instead of creating a String for every element. https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=af1a94b06ea1a2126606b202862ecfc7

Reply:
great point!